asset/asset.go

```diff
@@ -953,21 +954,49 @@ type GroupKeyReveal interface {
 // that includes the specified data. If the data is nil, the leaf will not
 // contain any data but will still be a valid non-spendable script leaf.
 //
-// The script leaf is made non-spendable by including an OP_RETURN opcode at the
-// start of the script. While the script can still be executed, it will always
-// fail and cannot be used to spend funds.
-func NewNonSpendableScriptLeaf(data []byte) (txscript.TapLeaf, error) {
-	// Construct a script builder and add the OP_RETURN opcode to the start
-	// of the script to ensure that the script is non-executable.
-	scriptBuilder := txscript.NewScriptBuilder().AddOp(txscript.OP_RETURN)
+// The script leaf is made non-spendable by including an OP_RETURN/OP_CHECKSIG
+// opcode at the start of the script. While the script can still be executed, it
+// will always fail and cannot be used to spend funds.
+func NewNonSpendableScriptLeaf(version GkrV1Version,
+	data []byte) (txscript.TapLeaf, error) {
+
+	var builder *txscript.ScriptBuilder
+	switch version {
+	// For the OP_RETURN based version, we'll use a single OP_RETURN opcode.
+	case OpReturnVersion:
+		builder = txscript.NewScriptBuilder().AddOp(txscript.OP_RETURN)
+		if data != nil {
+			builder = builder.AddData(data)
+		}
+
+	// For the Pedersen commitment based version, we'll use a single
+	// OP_CEHCKSIG with an un-spendable key.
+	case PedersenVersion:
+		var msg [sha256.Size]byte
+		copy(msg[:], data)
+
+		// Make a Pedersen opening that uses no mask (we don't carry on
+		// the random value, as we don't care about hiding here). We'll
+		// also use the existing NUMs point.
+		op := pedersen.Opening{
+			Msg: msg,
+		}
+		commitPoint := pedersen.NewCommitment(op).Point()
+
+		commitBytes := schnorr.SerializePubKey(&commitPoint)
+
+		builder = txscript.NewScriptBuilder().AddData(commitBytes).
+			AddOp(txscript.OP_CHECKSIG)
 
-	// Add the data to the script if it is provided.
-	if data != nil {
-		scriptBuilder.AddData(data)
+	default:
+		// TODO(roasbeef): auto accept of new version for forwards
+		// compat?
+		return txscript.TapLeaf{}, fmt.Errorf("unknown "+
+			"version %v", version)
```

Comment:
We need to ensure that the correct leaf script is constructed for the given version. What does "auto accept" mean here? I think returning an error is the behaviour we want.

Reply:
Yeah you're right, this was just a fleeting thought I had during implementation thinking about future/possible changes.